# DP: Disable D tests, hang on many buildds

Index: b/src/gcc/d/Make-lang.in
===================================================================
--- a/src/gcc/d/Make-lang.in
+++ b/src/gcc/d/Make-lang.in
@@ -284,6 +284,6 @@ d.stagefeedback: stagefeedback-start
 # check targets.  However, our DejaGNU framework requires 'check-gdc' as its
 # entry point.  We feed the former to the latter here.
 check-d: check-gdc
-lang_checks += check-gdc
-lang_checks_parallelized += check-gdc
-check_gdc_parallelize = 10
+#lang_checks += check-gdc
+#lang_checks_parallelized += check-gdc
+#check_gdc_parallelize = 10
