Index: b/src/libphobos/m4/druntime.m4
===================================================================
--- a/src/libphobos/m4/druntime.m4
+++ b/src/libphobos/m4/druntime.m4
@@ -78,7 +78,7 @@ AC_DEFUN([DRUNTIME_INSTALL_DIRECTORIES],
   AC_SUBST(toolexeclibdir)
 
   # Default case for install directory for D sources files.
-  gdc_include_dir='${libdir}/gcc/${target_alias}'/${d_gcc_ver}/include/d
+  gdc_include_dir='${libdir}/gcc-cross/${target_alias}'/${d_gcc_ver}/include/d
   AC_SUBST(gdc_include_dir)
 ])
 
