Index: b/src/boehm-gc/configure.host
===================================================================
--- a/src/boehm-gc/configure.host
+++ b/src/boehm-gc/configure.host
@@ -41,7 +41,7 @@ else
 fi
 
 case "${host}" in
-  *-linux*|*-kfreebsd*-gnu*)
+  *-linux*|*-kfreebsd*-gnu*|*-gnu*)
     gc_use_mmap=yes
     ;;
 esac
Index: b/src/boehm-gc/include/private/gcconfig.h
===================================================================
--- a/src/boehm-gc/include/private/gcconfig.h
+++ b/src/boehm-gc/include/private/gcconfig.h
@@ -2132,7 +2132,7 @@
 #   endif
 # endif
 
-#if defined(LINUX) && defined(USE_MMAP)
+#if (defined(LINUX) || defined(HURD)) && defined(USE_MMAP)
     /* The kernel may do a somewhat better job merging mappings etc.	*/
     /* with anonymous mappings.						*/
 #   define USE_MMAP_ANON
