# DP: Proposed patch for PR sanitizer/67899

--- a/src/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
+++ b/src/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
@@ -564,13 +564,12 @@
 #else
     __sanitizer_sigset_t sa_mask;
 #ifndef __mips__
-#if defined(__sparc__)
-    unsigned long sa_flags;
-#else
+#if defined(__sparc__) && defined(__arch64__)
+    int __pad;
+#endif
     int sa_flags;
 #endif
 #endif
-#endif
 #if SANITIZER_LINUX
     void (*sa_restorer)();
 #endif
@@ -593,7 +592,8 @@
       void (*handler)(int signo);
       void (*sigaction)(int signo, void *info, void *ctx);
     };
-    unsigned long sa_flags;
+    int __pad;
+    int sa_flags;
     void (*sa_restorer)(void);
     __sanitizer_kernel_sigset_t sa_mask;
   };
