# DP: On sparc default to ultrasparc (v9a) in 32bit mode

Index: b/src/gcc/config.gcc
===================================================================
--- a/src/gcc/config.gcc
+++ b/src/gcc/config.gcc
@@ -4369,6 +4369,20 @@ do
 			;;
 		esac
 
+		# setting this directly fails to build a biarch defaulting to 32bit
+		case "$target" in
+		    sparc-*-linux*)
+			if test "$option" = cpu; then
+				val=ultrasparc
+			fi
+			;;
+		    DISABLEDsparc64-*-linux*)
+			if test "$option" = cpu_32; then
+				val=ultrasparc
+			fi
+			;;
+		esac
+
 		if test "x$t" = x
 		then
 			t="{ \"$option\", \"$val\" }"
