# DP: Disable D tests, hang on many buildds

--- a/src/gcc/d/Make-lang.in
+++ b/src/gcc/d/Make-lang.in
@@ -358,8 +358,8 @@
 # entry point.  We feed the former to the latter here.
 check-d: check-gdc
 # List of targets that can use the generic check- rule and its // variant.
-lang_checks += check-gdc
-lang_checks_parallelized += check-gdc
+#lang_checks += check-gdc
+#lang_checks_parallelized += check-gdc
 # For description see comment above check_gcc_parallelize in gcc/Makefile.in.
 check_gdc_parallelize = d_do_test.exp=runnable/*
 
