Index: gcc-4.8-4.8.1/src/gcc/config/rs6000/t-linux64
===================================================================
--- gcc-4.8-4.8.1.orig/src/gcc/config/rs6000/t-linux64	2013-07-29 07:09:15.537646145 +0000
+++ gcc-4.8-4.8.1/src/gcc/config/rs6000/t-linux64	2013-07-29 07:12:04.297649530 +0000
@@ -25,15 +25,24 @@
 # it doesn't tell anything about the 32bit libraries on those systems.  Set
 # MULTILIB_OSDIRNAMES according to what is found on the target.
 
-MULTILIB_OPTIONS        = m64/m32
-MULTILIB_DIRNAMES       = 64 32
-MULTILIB_EXTRA_OPTS     = fPIC
-ifneq (,$(findstring powerpc64,$(target)))
-MULTILIB_OSDIRNAMES	= ../lib$(call if_multiarch,:powerpc64-linux-gnu)
-MULTILIB_OSDIRNAMES    += ../lib32$(call if_multiarch,:powerpc-linux-gnu)
-else
-MULTILIB_OSDIRNAMES	= ../lib64$(call if_multiarch,:powerpc64-linux-gnu)
-MULTILIB_OSDIRNAMES    += ../lib$(call if_multiarch,:powerpc-linux-gnu)
-endif
+ifeq (,$(findstring powerpc64le,$(target)))
+  MULTILIB_OPTIONS        = m64/m32
+  MULTILIB_DIRNAMES       = 64 32
+  MULTILIB_EXTRA_OPTS     = fPIC
+  ifneq (,$(findstring powerpc64,$(target)))
+  MULTILIB_OSDIRNAMES     = ../lib$(call if_multiarch,:powerpc64-linux-gnu)
+  MULTILIB_OSDIRNAMES    += ../lib32$(call if_multiarch,:powerpc-linux-gnu)
+  else
+  MULTILIB_OSDIRNAMES     = ../lib64$(call if_multiarch,:powerpc64-linux-gnu)
+  MULTILIB_OSDIRNAMES    += ../lib$(call if_multiarch,:powerpc-linux-gnu)
+  endif
+
+  MULTIARCH_DIRNAME = $(call if_multiarch,powerpc$(if $(findstring 64,$(target)),64)-linux-gnu)
 
-MULTIARCH_DIRNAME = $(call if_multiarch,powerpc$(if $(findstring 64,$(target)),64)-linux-gnu)
+else 
+  MULTILIB_OPTIONS        =
+  MULTILIB_DIRNAMES       =
+  MULTILIB_EXTRA_OPTS     =
+  MULTILIB_OSDIRNAMES     =
+  MULTIARCH_DIRNAME = $(call if_multiarch,powerpc64le-linux-gnu)
+endif
