# DP: Proposed patch for PR c/57653.

Index: b/src/gcc/c-family/c-opts.c
===================================================================
--- a/src/gcc/c-family/c-opts.c
+++ b/src/gcc/c-family/c-opts.c
@@ -1363,6 +1363,10 @@ c_finish_options (void)
 static void
 push_command_line_include (void)
 {
+  // This can happen if disabled by -imacros for example.
+  if (include_cursor > deferred_count)
+    return;
+
   if (!done_preinclude)
     {
       done_preinclude = true;
