# DP: Install libjava libraries to multiarch location

--- a/src/libjava/classpath/m4/acinclude.m4
+++ b/src/libjava/classpath/m4/acinclude.m4
@@ -252,6 +252,10 @@
     .) toolexeclibdir=${libdir} ;; # Avoid trailing /.
     *) toolexeclibdir=${libdir}/${multi_os_directory} ;;
   esac
+  multiarch=`$CC -print-multiarch`
+  if test -n "$multiarch"; then
+    toolexeclibdir=${libdir}/${multiarch}
+  fi
   AC_SUBST(toolexeclibdir)
 ])
 
--- a/src/libjava/configure.ac
+++ b/src/libjava/configure.ac
@@ -1557,6 +1557,10 @@
       ../lib*) toolexeclibdir='$(subst /lib/../lib,/lib,'$toolexecmainlibdir/$multi_os_directory')' ;;
       *) toolexeclibdir=$toolexecmainlibdir/$multi_os_directory ;;
     esac
+    multiarch=`$CC -print-multiarch`
+    if test -n "$multiarch"; then
+        toolexeclibdir=$toolexecmainlibdir/$multiarch
+    fi
     ;;
 esac
 AC_SUBST(toolexecdir)
@@ -1583,6 +1587,10 @@
    dbexecdir='$(libdir)/'$multi_os_directory/$gcjsubdir
    ;;
 esac
+multiarch=`$CC -print-multiarch`
+if test -n "$multiarch"; then
+    dbexecdir='$(libdir)/'$multiarch/$gcjsubdir
+fi
 AC_SUBST(dbexecdir)
 AC_SUBST(gcjsubdir)
 
--- a/src/libjava/Makefile.am
+++ b/src/libjava/Makefile.am
@@ -334,7 +334,7 @@
 	-DGCJ_VERSIONED_LIBDIR="\"$(dbexecdir)\"" \
 	-DPATH_SEPARATOR="\"$(CLASSPATH_SEPARATOR)\"" \
 	-DECJ_JAR_FILE="\"$(ECJ_JAR)\"" \
-	-DLIBGCJ_DEFAULT_DATABASE="\"/var/lib/gcj-4.5/$(db_name)\"" \
+	-DLIBGCJ_DEFAULT_DATABASE="\"/var/lib/$(MULTIARCH_DIR)/gcj-4.5/$(db_name)\"" \
 	-DLIBGCJ_DEFAULT_DATABASE_PATH_TAIL="\"$(db_pathtail)\""
 
 AM_GCJFLAGS = \
--- a/src/libjava/Makefile.in
+++ b/src/libjava/Makefile.in
@@ -1105,7 +1105,7 @@
 	-DGCJ_VERSIONED_LIBDIR="\"$(dbexecdir)\"" \
 	-DPATH_SEPARATOR="\"$(CLASSPATH_SEPARATOR)\"" \
 	-DECJ_JAR_FILE="\"$(ECJ_JAR)\"" \
-	-DLIBGCJ_DEFAULT_DATABASE="\"/var/lib/gcj-4.5/$(db_name)\"" \
+	-DLIBGCJ_DEFAULT_DATABASE="\"/var/lib/$(MULTIARCH_DIR)/gcj-4.5/$(db_name)\"" \
 	-DLIBGCJ_DEFAULT_DATABASE_PATH_TAIL="\"$(db_pathtail)\""
 
 AM_GCJFLAGS = \
