--- a/src/libgcc/config/rs6000/crtsavfpr.S
+++ b/src/libgcc/config/rs6000/crtsavfpr.S
@@ -33,6 +33,7 @@
 
 /* On PowerPC64 Linux, these functions are provided by the linker.  */
 #ifndef __powerpc64__
+#ifndef __NO_FPRS__
 
 /* Routines for saving floating point registers, called by the compiler.  */
 /* Called with r11 pointing to the stack header word of the caller of the */
@@ -79,3 +80,4 @@
 CFI_ENDPROC
 
 #endif
+#endif
--- a/src/libgcc/config/rs6000/crtresfpr.S
+++ b/src/libgcc/config/rs6000/crtresfpr.S
@@ -33,6 +33,7 @@
 
 /* On PowerPC64 Linux, these functions are provided by the linker.  */
 #ifndef __powerpc64__
+#ifndef __NO_FPRS__
 
 /* Routines for restoring floating point registers, called by the compiler.  */
 /* Called with r11 pointing to the stack header word of the caller of the */
@@ -79,3 +80,4 @@
 CFI_ENDPROC
 
 #endif
+#endif
--- a/src/libgcc/config/rs6000/crtresxfpr.S
+++ b/src/libgcc/config/rs6000/crtresxfpr.S
@@ -33,6 +33,7 @@
 
 /* On PowerPC64 Linux, these functions are provided by the linker.  */
 #ifndef __powerpc64__
+#ifndef __NO_FPRS__
 
 /* Routines for restoring floating point registers, called by the compiler.  */
 /* Called with r11 pointing to the stack header word of the caller of the */
@@ -124,3 +125,4 @@
 CFI_ENDPROC
 
 #endif
+#endif
--- a/src/libgcc/config/rs6000/crtsavevr.S	2013-03-13 22:25:25.802681336 +0000
+++ b/src/libgcc/config/rs6000/crtsavevr.S	2013-03-13 22:26:21.054695066 +0000
@@ -24,6 +24,7 @@
 
 /* On PowerPC64 Linux, these functions are provided by the linker.  */
 #ifndef __powerpc64__
+#ifndef __NO_FPRS__
 
 #undef __ALTIVEC__
 #define __ALTIVEC__ 1
@@ -85,3 +86,4 @@
 CFI_ENDPROC
 
 #endif
+#endif
--- a/src/libgcc/config/rs6000/crtrestvr.S	2013-03-13 22:25:28.394681980 +0000
+++ b/src/libgcc/config/rs6000/crtrestvr.S	2013-03-13 22:26:21.058695067 +0000
@@ -24,6 +24,7 @@
 
 /* On PowerPC64 Linux, these functions are provided by the linker.  */
 #ifndef __powerpc64__
+#ifndef __NO_FPRS__
 
 #undef __ALTIVEC__
 #define __ALTIVEC__ 1
@@ -85,3 +86,4 @@
 CFI_ENDPROC
 
 #endif
+#endif
