--- src.orig/os-linux.h	2004-11-04 00:08:07.000000000 +0300
+++ src/os-linux.h	2005-01-04 22:56:46.000000000 +0300
@@ -26,6 +26,8 @@
 #ifndef os_linux_h
 #define os_linux_h
 
+#include <linux/limits.h>
+
 struct map_iterator
   {
     off_t offset;
--- src.orig/ptrace/_UPT_find_proc_info.c	2004-11-04 00:08:07.000000000 +0300
+++ src/ptrace/_UPT_find_proc_info.c	2005-01-04 23:37:34.000000000 +0300
@@ -35,6 +35,7 @@
 #if UNW_TARGET_IA64
 
 #include "elf64.h"
+#include <linux/limits.h>
 
 static unw_word_t
 find_gp (struct UPT_info *ui, Elf64_Phdr *pdyn, Elf64_Addr load_base)
