# DP: Fix gnat build on mips64el

--- a/src/gcc/ada/gcc-interface/Makefile.in
+++ b/src/gcc/ada/gcc-interface/Makefile.in
@@ -1893,7 +1893,7 @@ ifeq ($(strip $(filter-out mips64el linu
   LIBGNAT_TARGET_PAIRS_64 = \
   system.ads<system-linux-mips64el.ads
 
-  ifeq ($(strip $(shell $(GCC_FOR_TARGET) $(GNATLIBCFLAGS) -print-multi-os-directory)),../lib64)
+  ifneq ($(strip $(MULTISUBDIR)),/32)
     LIBGNAT_TARGET_PAIRS = \
     $(LIBGNAT_TARGET_PAIRS_COMMON) $(LIBGNAT_TARGET_PAIRS_64)
   else
