DP: Improve support for ZCX on arm.

Index: b/src/gcc/ada/gcc-interface/Makefile.in
===================================================================
--- a/src/gcc/ada/gcc-interface/Makefile.in
+++ b/src/gcc/ada/gcc-interface/Makefile.in
@@ -1921,7 +1921,10 @@ ifeq ($(strip $(filter-out arm% linux-gn
     LIBGNAT_TARGET_PAIRS += \
     system.ads<system-linux-armeb.ads
   else
-    EH_MECHANISM=-arm
+    # Special case; the GCC exception mechanism is supported under
+    # another name and with different files than for other
+    # target_cpus.
+    override EH_MECHANISM=-arm
     LIBGNAT_TARGET_PAIRS += \
     system.ads<system-linux-armel.ads \
     a-exexpr.adb<a-exexpr-gcc.adb \
