# DP: Proposed patch for PR sanitizer/67899

Index: b/src/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
===================================================================
--- a/src/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
+++ b/src/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
@@ -406,11 +406,10 @@ namespace __sanitizer {
       void (*sa_sigaction)(int sig, void *siginfo, void *uctx);
     };
     __sanitizer_sigset_t sa_mask;
-#if defined(__sparc__)
-    unsigned long sa_flags;
-#else
-    int sa_flags;
+#if defined(__sparc__) && defined(__arch64__)
+    int __pad;
 #endif
+    int sa_flags;
 #if SANITIZER_LINUX
     void (*sa_restorer)();
 #endif
@@ -425,7 +424,8 @@ namespace __sanitizer {
       void (*sigaction)(int signo, void *info, void *ctx);
       void (*handler)(int signo);
     };
-    unsigned long sa_flags;
+    int __pad;
+    int sa_flags;
     void (*sa_restorer)(void);
     __sanitizer_kernel_sigset_t sa_mask;
   };
