# DP: Set the D target include directory to a multiarch location.

--- a/src/gcc/d/Make-lang.in
+++ b/src/gcc/d/Make-lang.in
@@ -61,7 +61,11 @@
 	$(D_DMD_H)
 
 
-gcc_d_target_include_dir=$(gcc_d_include_dir)/$(target_noncanonical)
+ifneq (,$(MULTIARCH_DIRNAME))
+  gcc_d_target_include_dir = /usr/include/$(MULTIARCH_DIRNAME)/d/$(version)
+else
+  gcc_d_target_include_dir=$(gcc_d_include_dir)/$(target_noncanonical)
+endif
 
 # Name of phobos library
 D_LIBPHOBOS = -DLIBPHOBOS=\"gphobos2\"
